tools/logger/convert.c

```diff
@@ -200,6 +200,13 @@ static void process_params(struct proc_ldc_entry *pe,
 		if (p[1] == '%') {
 			/* Skip "%%" */
 			p += 2;
+		} else if (p[1] == 's') {
+			/* check for string printing, because it leads to logger crash */
+			log_err("String printing is not supported\n");
+			pe->params[i] = (uintptr_t)asprintf("<String @ 0x%08x>", e->params[i]);
```

Comment:
Acceptable memory leak?

Reply:
subst_mask is set, so there shouldn't be memory leak

Reply:
Oh, so you use subst_mask to clean this up?

Reply:
It has been implemented before,
```
static void free_proc_ldc_entry(struct proc_ldc_entry *pe)
{
	int i;

	for (i = 0; i < TRACE_MAX_PARAMS_COUNT; i++) {
		if (pe->subst_mask & (1 << i))
			free((void *)pe->params[i]);
		pe->params[i] = 0;
	}
}
```